lib/http/Http.ts

```diff
@@ -117,13 +122,27 @@ export class Http {
   }
 
   async upload<T = any>(url: string, body?: any, options?: FetchOptions): Promise<T> {
-    const formData = this.objectToFormData(body)
+    return this.post(url, this.objectToFormData(body), options)
+  }
 
-    return this.performRequest<T>(url, {
-      method: 'POST',
-      body: formData,
-      ...options
+  async postOrUpload<T = any>(url: string, body?: any, options?: FetchOptions): Promise<T> {
```

Comment:
maybe name this something else, or merge with Http.post behind an option 👀

Reply:
Yeah I was thinking to say that too 😆 

I think it's good to have different method since it does something special.

How about... `multipartJson`...? (OK C'mon I don't like it too! It's your turn!

Reply:
GPT ideas:

```
- postWithFileHandling
- postWithUploadSupport
- postWithMultipart
- postWithFileAwareness
- enhancedPost
- multipartPost
- fileAwarePost
- uploadPost
- formDataPost
- postWithFileSupport
- postWithFormData
- postWithUploadHandling
```

Reply:
The naming is determined by its intended use. It should coincide with how it’s interpreted in the backend because of its exclusivity. `postOrUpload` suggests one or the other, when in reality it could be both. I’m assuming this method is to accommodate files while stringifying the adjacent payload?

`postWithPayload` or `postWithAttachment` come to mind which acknowledges the distinction of the formatted payload.

The alternative is to merge with `post` itself with extended options, and make use of backend middleware which can gracefully resolve the contents from the given key.